Mathlib/Analysis/InnerProductSpace/GramMatrix.lean

```diff
@@ -0,0 +1,123 @@
+/-
+Copyright (c) 2025 Peter Pfaffelhuber. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Peter Pfaffelhuber
+-/
+
+import Mathlib.LinearAlgebra.Matrix.PosDef
+
+/-! # Gram Matrices
+
+This file defines Gram matrices and proves their positive semi-definiteness.
+Results require `RCLike 𝕜`.
+
+## Main definition
+
+* `Matrix.gram` : the `Matrix n n 𝕜` with `⟪v i, v j⟫` at `i j : n`, where `v : n → E` for an
+  `Inner 𝕜 E`.
+
+## Main results
+
+* `Matrix.gram_posSemidef` Gram matrices are positive semi-definite.
+-/
+
+open RCLike Real Matrix
+
+open scoped InnerProductSpace
+
+open scoped ComplexOrder
+
+variable {E n : Type*}
+variable {α : Type*}
+variable {𝕜 : Type*}
+namespace Matrix
+
+/-- The entries of a Gram matrix are inner products of vectors in an inner product space. -/
+def gram (𝕜 : Type*) [Inner 𝕜 E] (v : n → E) : Matrix n n 𝕜 := of fun i j ↦ inner 𝕜 (v i) (v j)
+
+local notation "⟪" x ", " y "⟫" => @inner 𝕜  _ _ x y
+
+lemma gram_apply [Inner 𝕜 E] {v : n → E} (i j : n) :
+    (gram _ v) i j = ⟪v i, v j⟫ := by
+  rfl
```

Comment:
`v` should be explicithere, and you may as well squeeze to one line and drop the `by`
```suggestion
lemma gram_apply [Inner 𝕜 E] (v : n → E) (i j : n) : (gram _ v) i j = ⟪v i, v j⟫ := rfl
```